custom_components/adaptive_lighting/switch.py

```diff
@@ -238,57 +237,136 @@ def _split_service_data(service_data, adapt_brightness, adapt_color):
+                "pass a switch under 'entity_id'. See the readme for details. Got %s",
+                service_call.data,
+            )
+            raise ValueError(
+                "adaptive-lighting: Light %s not found in any switch's configuration.",
+                lights,
+            )
+    # Check assumedly not needed but exists for testing.
+    if not switch:  # Probably not needed but we're including this during testing.
+        _LOGGER.error(
+            "bad service data to adaptive-lighting service call - "
+            "entities were not found in integration. Service data:\n%s",
+            service_call.data,
+        )
+        raise ValueError("adaptive-lighting: User sent incorrect data to service call")
+    _LOGGER.debug("Switches used: %s", switch)
+    return switch, data
 
 
-async def handle_set_manual_control(switch: AdaptiveSwitch, service_call: ServiceCall):
-    """Set or unset lights as 'manually controlled'."""
-    lights = service_call.data[CONF_LIGHTS]
-    if not lights:
-        all_lights = switch._lights  # pylint: disable=protected-access
-    else:
-        all_lights = _expand_light_groups(switch.hass, lights)
-    _LOGGER.debug(
-        "Called 'adaptive_lighting.set_manual_control' service with '%s'",
-        service_call.data,
+# From https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/template.py#L1109
+def integration_entities(hass: HomeAssistant, entry_name: str):
+    """Get entity ids for entities tied to an integration/domain.
+    Provide entry_name as domain to get all entity id's for a integration/domain
+    or provide a config entry title for filtering between instances of the same
+    integration.
+    """
+    # first try if this is a config entry match
+    conf_entry = next(
+        (
+            entry.entry_id
+            for entry in hass.config_entries.async_entries()
+            if entry.title == entry_name
+        ),
+        None,
     )
-    if service_call.data[CONF_MANUAL_CONTROL]:
-        for light in all_lights:
-            switch.turn_on_off_listener.manual_control[light] = True
-            _fire_manual_control_event(switch, light, service_call.context)
-    else:
-        switch.turn_on_off_listener.reset(*all_lights)
-        # pylint: disable=protected-access
-        if switch.is_on:
-            await switch._update_attrs_and_maybe_adapt_lights(
-                all_lights,
-                transition=switch._initial_transition,
-                force=True,
-                context=switch.create_context("service", parent=service_call.context),
-            )
+    if conf_entry is not None:
+        ent_reg = entity_registry.async_get(hass)
+        entries = entity_registry.async_entries_for_config_entry(ent_reg, conf_entry)
+        return [entry.entity_id for entry in entries]
+
+    # fallback to just returning all entities for a domain
+    # pylint: disable-next=import-outside-toplevel
+    from .entity import entity_sources
```

Comment:
Should we just import this at the top?

Reply:
see https://github.com/basnijholt/adaptive-lighting/pull/459/commits/aeed1da0b561e3c2b0aead1d02aab90f930afb7f